packages/sdk-client-v3/src/utils/executor.ts

```diff
@@ -68,17 +70,31 @@ export default async function executor(request: HttpClientConfig) {
       }
 
       async function executeWithRetry<T = any>(): Promise<T> {
-        // first attempt
-        let _response = await execute()
-
-        if (predicate(retryCodes, _response)) return _response
+        const executeWithTryCatch = async (retryCodes) => {
+          let _response = {} as any
+          try {
+            _response = await execute()
+            if (predicate(retryCodes, _response)) return _response
+          } catch (e) {
+            if (e.name.includes('AbortError')) {
+              _response = e
+            } else {
+              throw e
```

Comment:
I think we need a test case here, to how the SDK handles this exception, hopefully in an integration test.

Reply:
https://github.com/commercetools/commercetools-sdk-typescript/pull/714/commits/f435d6bd4df8dcc9b958d49fb2bb0199ed14d095

I added only a unit test. I can't find a reproduceable case when we would get this error with integration test. If you have a suggestion, let me know.

Reply:
Sure, we can use the signal property to signal the AbortController to abort the request at any time using the abort timer. See my commend below.